src/transformers/generation/utils.py

```diff
@@ -1421,6 +1421,11 @@ def generate(
                 )
             generation_config.max_length = generation_config.max_new_tokens + input_ids_length
 
+        # adjust max_length when using `input_embeds` in decoder-only models
+        elif len(inputs_tensor.shape) == 3 and inputs_tensor.shape[:-1] != input_ids.shape:
+            if not self.config.is_encoder_decoder:
+                generation_config.max_length -= inputs_tensor.shape[1] - 1
```

Comment:
this -1 to the `decoder_start_id` so that at the end the decoded text is same, without special_tokens